packages/@aws-cdk/cloudformation-include/test/nested-stacks.test.ts

```diff
@@ -0,0 +1,528 @@
+    const anotherChildStack = parentTemplate.getNestedStack('AnotherChildStack');
+    expect(childStack.stack).toMatchTemplate(
+      loadTestFileToJsObject('grandchild-stack.expected.json'),
+    );
+
+    expect(anotherChildStack.stack).toMatchTemplate(
+      loadTestFileToJsObject('grandchild-stack.expected.json'),
+    );
+  });
+
+  test('can ingest a template with one child and one grandchild', () => {
+    const parentTemplate = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-export-stack.json'),
+      nestedStacks: {
+        'ChildStack': {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+          nestedStacks: {
+            'GrandChildStack': {
+              templateFile: testTemplateFilePath('grandchild-import-stack.json'),
+            },
+          },
+        },
+      },
+    });
+
+    const childStack = parentTemplate.getNestedStack('ChildStack');
+    const grandChildStack = childStack.includedTemplate.getNestedStack('GrandChildStack');
+    expect(childStack.stack).toMatchTemplate(
+      loadTestFileToJsObject('child-stack.expected.json'),
+    );
+
+    expect(grandChildStack.stack).toMatchTemplate(
+      loadTestFileToJsObject('grandchild-stack.expected.json'),
+    );
+  });
+
+  test('throws an error when provided a nested stack that is not present in the template', () => {
+    expect(() => {
+      new inc.CfnInclude(stack, 'ParentStack', {
+        templateFile: testTemplateFilePath('parent-export-stack.json'),
+        nestedStacks: {
+          'FakeStack': {
+            templateFile: testTemplateFilePath('child-import-stack.json'),
+          },
+        },
+      });
+    }).toThrow(/Nested Stack with logical ID 'FakeStack' was not found in the template/);
+  });
+
+  test('Throws an error when NestedStacks contains an ID that is not a CloudFormation::Stack in the template', () => {
+    expect(() => {
+      new inc.CfnInclude(stack, 'ParentStack', {
+        templateFile: testTemplateFilePath('parent-export-stack.json'),
+        nestedStacks: {
+          'ChildStack': {
+            templateFile: testTemplateFilePath('child-import-stack.json'),
+            nestedStacks: {
+              'BucketImport': {
+                templateFile: testTemplateFilePath('grandchild-import-stack.json'),
+              },
+            },
+          },
+        },
+      });
+    }).toThrow(/Nested Stack with logical ID 'BucketImport' is not an AWS::CloudFormation::Stack resource/);
+  });
+
+  test('can modify resources in nested stacks', () => {
+    const parent = new inc.CfnInclude(stack, 'ParentStack', {
+      templateFile: testTemplateFilePath('parent-export-stack.json'),
+      nestedStacks: {
+        'ChildStack': {
+          templateFile: testTemplateFilePath('child-import-stack.json'),
+          nestedStacks: {
+            'GrandChildStack': {
+              templateFile: testTemplateFilePath('grandchild-import-stack.json'),
```

Comment:
Again, I don't see why we need to go all the way to grandchildren to write this simple tests, but OK.